core/src/main/scala/io/finch/request/package.scala

```diff
@@ -49,14 +129,22 @@ package object request {
     def apply[Req](req: Req)(implicit ev: Req => HttpRequest): Future[A]
 
     def flatMap[B](fn: A => RequestReader[B]) = new RequestReader[B] {
+      val item = MultipleItems
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) flatMap { fn(_)(req) }
     }
 
     def map[B](fn: A => B) = new RequestReader[B] {
+      val item = self.item
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) = self(req) map fn
     }
     
+    def mapFuture[B](fn: A => Future[B]) = new RequestReader[B] {
```

Comment:
It's a good question. it 5am, and I should probably be asleep. I'll have a think, but I am sure @jenshalm can come up with a nice name. He has a talent for that kind of thing.